src/lib/actions/RulesAction.ts

```diff
@@ -1,7 +1,43 @@
+import {CodeAnalyzer, CodeAnalyzerConfig, Rule, RuleSelection} from '@salesforce/code-analyzer-core';
 
+import {ConfigLoader} from '../loaders/ConfigLoader';
+import {EngineLoader} from '../loaders/EngineLoader';
+import {RuleViewer} from '../viewers/RuleViewer';
+
+export type RulesDependencies = {
+	configLoader: ConfigLoader;
+	engineLoader: EngineLoader;
+	viewer: RuleViewer;
+}
+
+export type RulesInput = {
+	'config-file'?: string;
+	'rule-selector': string[];
+}
 
 export class RulesAction {
-	public execute(_input: object): void {
+	public execute(dependencies: RulesDependencies, input: RulesInput): void {
+		// Step 1: Get the config.
```

Comment:
nitpick: generally I would avoid comments that specify "what" is being done and rather only focus on "why". But since this is the first command that is implementing the new design, the comments were in fact useful :) So, I'm on the fence about these.